core/kernel/ldelf_syscalls.c

```diff
@@ -136,16 +136,34 @@ TEE_Result ldelf_syscall_open_bin(const TEE_UUID *uuid, size_t uuid_size,
 	if (!binh)
 		return TEE_ERROR_OUT_OF_MEMORY;
 
-	SCATTERED_ARRAY_FOREACH(binh->op, ta_stores, struct ts_store_ops) {
-		DMSG("Lookup user TA ELF %pUl (%s)",
-		     (void *)uuid, binh->op->description);
-
-		res = binh->op->open(uuid, &binh->h);
-		DMSG("res=%#"PRIx32, res);
-		if (res != TEE_ERROR_ITEM_NOT_FOUND &&
-		    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
-			break;
+	if (is_user_ta_ctx(sess->ctx) || is_stmm_ctx(sess->ctx)) {
+		SCATTERED_ARRAY_FOREACH(binh->op, ta_stores,
+					struct ts_store_ops) {
+			DMSG("Lookup user TA ELF %pUl (%s)",
+			     (void *)uuid, binh->op->description);
+
+			res = binh->op->open(uuid, &binh->h);
+			DMSG("res=%#"PRIx32, res);
+			if (res != TEE_ERROR_ITEM_NOT_FOUND &&
+			    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
+				break;
+		}
+	} else if (is_sp_ctx(sess->ctx)) {
+		SCATTERED_ARRAY_FOREACH(binh->op, sp_stores,
+					struct ts_store_ops) {
+			DMSG("Lookup user SP ELF %pUl (%s)",
+			     (void *)uuid, binh->op->description);
+
+			res = binh->op->open(uuid, &binh->h);
+			DMSG("res=0x%x", res);
```

Comment:
`"res=%#"PRIx32`
Please fix the other places in this patch were a `TEE_Result` is printed.

Reply:
Done

Reply:
Not here. :-)

Reply:
Good catch :-). I have updated it